ts/lib/auth/computeclient.ts

```diff
@@ -39,9 +39,11 @@ export class Compute extends OAuth2Client {
    *
    * Retrieve access token from the metadata server.
    * See: https://developers.google.com/compute/docs/authentication
+   *
+   * @param {number=} refreshTokeEarlyMillis The token should be refreshed if it will expire within this many milliseconds.
    */
-  constructor() {
-    super();
+  constructor(refreshTokenEarlyMillis?: number) {
+    super(undefined, undefined, undefined, undefined, refreshTokenEarlyMillis);
```

Comment:
Oh my goodness. This doesn't have too great a readability.

I think the deficiency is in the constructor for `OAuth2Client`. Instead of taking so many optional properties as function parameters it should instead accept an options object. This would be a semver major change however, but perhaps it is worth making before 1.0? @JustinBeckwith WDYT?

Reply:
actually there might be a way to avoid a semver major here. If arguments.length is 1 and it is an object that has one of the expected properties, we assume the 'new form'. Else we use the existing positional parameters form.

WDYT?

I still slightly favor a breaking change given the chance with 1.0.

Reply:
I'm almost always in favor of accepting an options object over a mishmash of optional properties.  I would rather break it, and provide really nice error messages if we're going to do it.